plugins/arkworks/src/poseidon/config.rs

```diff
@@ -0,0 +1,234 @@
+//! Poseidon Configuration
+
+use crate::{
+    constraint::{fp::Fp, FpVar},
+    ff::Field,
+    poseidon::{Compiler, Specification},
+    r1cs_std::fields::FieldVar,
+};
+use eclair::alloc::Constant;
+use openzl_crypto::poseidon::{self, ParameterFieldType};
+
+/// Bn254 Scalar Field
+pub type ConstraintField = ark_bn254::Fr;
```

Comment:
Shouldn't this be elliptic curve agnostic? After all, this is an arkworks plugin which should support all elliptic curves supported by arkworks

Reply:
In principle this would change the impls of `poseidon::Constants` depending on the EC. It appears that they're the same for Bn254 and BLS12-381 though.